### PR TITLE
V050

### PR DESCRIPTION
**Pull request by atayarani**
@Tapjoy/eng-group-ops 

Notes:
* I only wrote specs for the classes that I already refactored
* replaces #5

### CODE REVIEW

File: lib/tapjoy/ldap/user/delete.rb [hunk 1]
```diff
@@ -3,33 +3,49 @@ module LDAP
...
-          prompt = '>'
-          dn = "uid=#{ opts[:user] },ou=People,#{ Tapjoy::LDAP::client.basedn }"
-          unless opts[:force]
-            puts "Confirm that you want to delete user: #{ opts[:user] }"
-            print prompt
-            confirm = STDIN.gets.chomp().downcase
-            unless confirm.eql?('y') || confirm.eql?('yes')
-              abort("Deletion of #{ opts[:user] } aborted")
-            end
+        def dn
```
Comment (marcuswalser) [inline lib/tapjoy/ldap/user/delete.rb]: Any objection to calling this `distinguished_name`? Would be clearer.

File: lib/tapjoy/ldap/user/delete.rb [hunk 2]
```diff
@@ -3,33 +3,49 @@ module LDAP
...
+          @dn ||= "uid=#{opts[:user]},ou=#{ou},#{Tapjoy::LDAP::client.basedn}"
+        end
+
+        def confirm
+          puts "Confirm that you want to delete user: #{ opts[:user] }"
```
Comment (marcuswalser) [inline lib/tapjoy/ldap/user/delete.rb]: `"Confirm that you want to delete user (yes/no): #{ opts[:user] }"` ?

File: lib/tapjoy/ldap/user/delete.rb [hunk 3]
```diff
@@ -3,33 +3,49 @@ module LDAP
...
+          print '>'
+          confirm = STDIN.gets.chomp().downcase
+          unless confirm.eql?('y') || confirm.eql?('yes')
+            abort("Deletion of #{ opts[:user] } aborted")
           end
+        end
 
-          puts Tapjoy::LDAP::client.delete(dn)
+        def ou
```
Comment (marcuswalser) [inline lib/tapjoy/ldap/user/delete.rb]: `organizational_unit`?

File: lib/tapjoy/ldap/user/delete.rb (hunk 1 above)
Comment (atayarani) [inline lib/tapjoy/ldap/user/delete.rb]: fixed

File: lib/tapjoy/ldap/user/delete.rb (hunk 2 above)
Comment (atayarani) [inline lib/tapjoy/ldap/user/delete.rb]: fixed

File: lib/tapjoy/ldap/user/delete.rb (hunk 3 above)
Comment (atayarani) [inline lib/tapjoy/ldap/user/delete.rb]: fixed

File: lib/tapjoy/ldap/group/delete.rb [hunk 4]
```diff
@@ -20,8 +20,8 @@ def opts
           end
         end
 
-        def dn
-          @dn ||= "cn=#{opts[:name]},ou=Group,#{Tapjoy::LDAP::client.basedn}"
+        def distinguished_name
+          @distinguished_name ||= "cn=#{opts[:name]},ou=Group,#{Tapjoy::LDAP::client.basedn}"
         end
 
         def confirm
```
Comment (ehealy) [inline lib/tapjoy/ldap/group/delete.rb]: not part of the PR but should the next line be 

```
puts "Confirm that you want to delete group #{opts[:name]} (yes/no)"
```

instead of 
```
puts "Confirm that you want to delete group #{opts[:group]} (yes/no)"
```

Comment (atayarani) [inline lib/tapjoy/ldap/group/delete.rb]: good catch.  Fixed.

File: spec/tapjoy/ldap/group/add_user_spec.rb [hunk 5]
```diff
@@ -0,0 +1,12 @@
+require_relative '../../../spec_helper'
+require_relative '../group'
+describe 'Tapjoy::LDAP::Group::AddUser.add_user' do
+  include_context 'group'
+  let(:operations) {[[:add, :memberUid, 'test.user']]}
+
+  it 'creates a group' do
```
Comment (ehealy) [inline spec/tapjoy/ldap/group/add_user_spec.rb]: 'adds a user to a group'

File: spec/tapjoy/ldap/group/delete_spec.rb [hunk 6]
```diff
@@ -0,0 +1,11 @@
+require_relative '../../../spec_helper'
+require_relative '../group'
+describe 'Tapjoy::LDAP::Group::Delete.delete' do
+  include_context 'group'
+
+  it 'creates a group' do
```
Comment (ehealy) [inline spec/tapjoy/ldap/group/delete_spec.rb]: 'deletes a group'

File: spec/tapjoy/ldap/group/add_user_spec.rb [hunk 7]
```diff
@@ -0,0 +1,12 @@
+require_relative '../../../spec_helper'
```
Comment (ehealy) [inline spec/tapjoy/ldap/group/add_user_spec.rb]: you can save some sanity by just doing require 'spec_helper' and require 'user' etc. in these. require should recurse outwards until it hits the file.

Comment (atayarani) [inline spec/tapjoy/ldap/group/add_user_spec.rb]: let's call that a TODO

File: spec/tapjoy/ldap/group/delete_spec.rb (hunk 6 above)
Comment (atayarani) [inline spec/tapjoy/ldap/group/delete_spec.rb]: fixed

File: spec/tapjoy/ldap/group/add_user_spec.rb (hunk 7 above)
Comment (atayarani) [inline spec/tapjoy/ldap/group/add_user_spec.rb]: fixed

File: spec/tapjoy/ldap/group/add_user_spec.rb (hunk 5 above)
Comment (atayarani) [inline spec/tapjoy/ldap/group/add_user_spec.rb]: fixed

**Comment by ehealy** [comment]
Specs are running clean and cleanups look sane. LGTM.
